test/clj_sqs_extended/core_test.clj

```diff
@@ -111,6 +114,61 @@
           (is (contains? stats :stopped-at))))
       (close! handler-chan))))
 
+(deftest handle-queue-terminates-after-restart-count-exceeded
+  (testing "handle-queue terminates when the restart-count exceeds the limit"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (with-redefs-fn {#'sqs/receive-message
+                         (fn [_ _ _] (HttpTimeoutException.
+                                       "Something seems to be wrong with the network."))}
+          #(let [stop-fn (fixtures/with-handle-queue-queue-opts-standard-no-autostop
+                           handler-chan
+                           {:restart-limit         3
+                            :restart-delay-seconds 1})]
+             (Thread/sleep 3500)
+             (let [stats (stop-fn)]
+               (is (= (:restart-count stats) 3))))))
+      (close! handler-chan))))
+
+(deftest handle-queue-restarts-after-recoverable-error-occured
+  (testing "handle-queue restarts properly and continues running upon recoverable error"
+    (let [handler-chan (chan)]
+      (fixtures/with-test-standard-queue
+        (let [stop-fn (fixtures/with-handle-queue-queue-opts-standard-no-autostop
+                        handler-chan
+                        {:restart-delay-seconds 1})]
+          ;; Cause some trouble in receive-message ...
+          (trouble/activate-troublemaker :receive-message)
+          (Thread/sleep 1500)
+
+          ;; All good again, back to work!
+          (trouble/deactivate-troublemaker :receive-message)
+          (Thread/sleep 1500)
```

Comment:
I see what you're trying to accomplish with this troublemaker hack here.

Thinking about how to rewrite this without using `troublemaker`... Perhaps redefs `sqs/receive-message` like you did above. But keep an internal atom on the redef'd `sqs/receive-message` with a counter. Such that after `sqs/receive-message` is called X times, raise an exception.

Pseudo code,

```clj
(fn [_ _ _] (let [called-counter (atom 0)
  (swap! called-counter inc)
  (when (> @called-counter 2)
    (HttpTimeoutException. "Testing network failure"))))
```

Reply:
This looks nice, I'll rework it according to your suggestion.